plugins/modules/ndo_tenant_igmp_interface_policy.py

```diff
@@ -0,0 +1,502 @@
+    group_timeout = module.params.get("group_timeout")
+    query_interval = module.params.get("query_interval")
+    query_response_interval = module.params.get("query_response_interval")
+    last_member_count = module.params.get("last_member_count")
+    last_member_response_time = module.params.get("last_member_response_time")
+    startup_query_count = module.params.get("startup_query_count")
+    startup_query_interval = module.params.get("startup_query_interval")
+    querier_timeout = module.params.get("querier_timeout")
+    robustness_variable = module.params.get("robustness_variable")
+    state_limit_route_map_uuid = module.params.get("state_limit_route_map_uuid")
+    state_limit_route_map = module.params.get("state_limit_route_map")
+    report_policy_route_map_uuid = module.params.get("report_policy_route_map_uuid")
+    report_policy_route_map = module.params.get("report_policy_route_map")
+    static_report_route_map_uuid = module.params.get("static_report_route_map_uuid")
+    static_report_route_map = module.params.get("static_report_route_map")
+    maximum_multicast_entries = module.params.get("maximum_multicast_entries")
+    reserved_multicast_entries = module.params.get("reserved_multicast_entries")
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "tenant", template)
+    mso_template.validate_template("tenantPolicy")
+    ops = []
+
+    existing_igmp_interface_policies = mso_template.template.get("tenantPolicyTemplate", {}).get("template", {}).get("igmpInterfacePolicies", [])
+    object_description = "IGMP Interface Policy"
+
+    if name or uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_igmp_interface_policies,
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+        if match:
+            igmp_interface_policy_attrs_path = "/tenantPolicyTemplate/template/igmpInterfacePolicies/{0}".format(match.index)
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_igmp_interface_policies
+
+    if state == "present":
+        if uuid and not mso.existing:
+            mso.fail_json(msg="{0} with the UUID: '{1}' not found".format(object_description, uuid))
+
+        if state_limit_route_map and not state_limit_route_map_uuid:
+            state_limit_route_map_uuid = mso_template.get_route_map_policy_for_multicast_uuid(state_limit_route_map.get("name"))
+
+        if report_policy_route_map and not report_policy_route_map_uuid:
+            report_policy_route_map_uuid = mso_template.get_route_map_policy_for_multicast_uuid(report_policy_route_map.get("name"))
+
+        if static_report_route_map and not static_report_route_map_uuid:
+            static_report_route_map_uuid = mso_template.get_route_map_policy_for_multicast_uuid(static_report_route_map.get("name"))
```

Comment:
static_report_route_map_uuid = mso_template.get_route_map_policy_for_multicast_uuid(static_report_route_map)

Reply:
This attribute is retained as dictionary as discussed.